packages/block-library/src/post-content/index.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Server-side rendering of the `core/post-content` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/post-content` block on the server.
+ *
+ * @return string Returns the filtered post content of the current post.
+ */
+function render_block_core_post_content() {
+	// TODO: Without this temporary fix, an infinite loop can occur.
+	if ( is_admin() || defined( 'REST_REQUEST' ) ) {
+		return '';
+	}
+	if ( ! in_the_loop() ) {
+		rewind_posts();
+		the_post();
+	}
```

Comment:
I noticed that this code is duplicated between post title and post content, As we discussed before, it seems to me that this code is better in a container "post" block. I'm ok moving forward without it and see where the limits of this approach lead us.

That said, we'll probably want to share that code across all post related blocks and avoid duplicating it.

Reply:
We'll see. I still think it's valuable to support top level post children blocks like these. The post wrapper block should only be used to overwrite the current post context.

Reply:
Can we extract the common code at least?

Reply:
https://github.com/WordPress/gutenberg/pull/18461/commits/d01218b8c9bf27423e4eff59b08d0b84e51bfe76